src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -142,6 +142,7 @@ private function handleRaw(Request $request, int $type = self::MAIN_REQUEST): Re
 
         // controller arguments
         $arguments = $this->argumentResolver->getArguments($request, $controller);
+        $request->attributes->remove('_controller_reflectors');
 
         $event = new ControllerArgumentsEvent($this, $controller, $arguments, $request, $type);
```

Comment:
I'm not sure I see what you mean: to me, the listener handling IsGranted will listen on ControllerEvent and call getAttributes on it. Why would ControllerArgumentsEvent need access to the attributes of the controller itself (compared to only the arguments')? To me, such a listener should listen to ControllerEvent.

Reply:
the IsGrantedListener runs on ControllerArgumentsEvent, not on ControllerEvent. The goal is to run *after* argument resolvers, so that you can vote on the resolved arguments.